### DISCUSSION TITLE
Zen Mode as A Text Editor?

### DISCUSSION THREAD

**Discussion by tajhans** [monkeytypegame/monkeytype#1687] [Ideas] [answered]
I think it would be cool (maybe not practical) to have the zen mode, or a similar version, to be used as a text editor. Like users could write essays and stuff. Maybe even to code? Maybe it just me but I could definitely see me using Monkeytype to write school essays and stuff. Any other ideas to add on?

The customization could be one of its main selling points. For example instead of a timer there would be a word count (important for essays) and of course people could change this for paragraph count, characters, etc. Also different ways to export like pdfs, word, and whatnot. 

The documents would be connected to the account, meaning that we would have to store that data somewhere, so more money for servers. It's not practical but it could be a cool idea to come back to.

**Comment by frogfrogfrogfrogfrog** [comment]
I'm pretty sure implementing an entire text editor like Google Docs wouldn't be viable but I could see a few features being put into Monkeytype but I think the most realistic thing any of the devs would do is saving the data in Zen mode

**Comment by Miodec** [comment, marked as answer]
As kevin mentioned, making a text editor doesn't seem viable, maybe not for now - it's pretty much a whole new project (Monkeyedit 👀). Currently there are other more important features to add.

**Reply by unrealapex** [reply]
Here's a poorly hacked together version I made a few weeks ago:
- [source](https://github.com/UnrealApex/compose) 
- [live site](https://compose.glitch.me/)

(trust me, I've made some better things, like [this prototype](https://beta-yt-player.glitch.me/) ,[ and this](https://github.com/UnrealApex/yt-player)):
